### ISSUE TITLE
[US] As MILES HARDIN, I want to enter the required information in the idea form so that I can submit my idea.

### ISSUE THREAD

**Issue by han1ya** [dev-launchers/dev-launchers-platform#662]
**Sub-tasks:**
- [x] UX design
- [x] Development
- [ ] QA

—----------------------------------------------------------------------------

**Considerations:**
- Include form validation (optional and mandatory fields) in idea submission page & idea workshopping pages.
- Max. Min. characters will be defined by @RETprojects  and @liwenlin664477 

Please refer to UX designs in [Figma ](https://www.figma.com/file/hEDIHsBZYa6hmK8As5mcjy/%5BDL---IdeaSpace%5D-UX-designs?node-id=2645%3A5947&t=8soOirvjw3YvrVlo-1) 

**Acceptance criteria:**
- Layout is responsive (break points are included in the Figma file)
- Users can only submit the form by filling in all the required fields. 
- If mandatory fields are not completed the user will be presented with an alert showing the missing input fields.

**Comment by JulieMass** [comment]
Hi @liwenlin664477 , @RETprojects _(CC @Nouri-Anouar + @pyxld-kris )_ I also discovered that our DL styleguide has inline input form validations apart from the error alerts (attaching a screen capture below). Maybe we can think of including this too to prevent user errors before clicking the "submit" button? 

Here is the Figma link so you can check out the components: https://www.figma.com/file/EwzuhhvTulvFRMvhTD5VAh/Dev-Launchers-Style-Guide?node-id=1055%3A1098

![image.png](https://images.zenhubusercontent.com/63400f3d1e79614e45d9ebd9/c3a66ec7-38ee-4d27-a639-aaa10ce766da)

**Comment by liwenlin664477** [comment]
> 
Sound good. I'll look into how to implement it.

**Comment by liwenlin664477** [comment]
@JulieMass Sound good. I'll look into how to implement it.

**Comment by liwenlin664477** [comment]
@JulieMass Hi Julie, I and Remi have already finished the functionality of this issue. I think we can move this issue into the Review/QA section. Thanks!

**Comment by JulieMass** [comment]
Hey Will, thanks for letting me know. This is excellent! great news Remi and Will. Moving the issue to Review/QA 🎉

**Comment by JulieMass** [comment]
Hi @RETprojects and @liwenlin664477 

(CC. @pyxld-kris )

I updated this user story's scope so now it should only cover form validations in the idea submission page and the idea workshopping page. My bad there! This pages detail wasn't included in the description and it might have contributed to the confusion 😔. Please let me know if the description is OK now or if I missed anything. 

I will try to be more detailed in future users' stories and I'll try to add screenshots too which might help us too. 

Finally I also created a separate enhancement task (#732) so we can develop in future sprints the new visual design that @Nouri-Anouar is reviewing.

**Comment by JulieMass** [comment]
(CC. @chadcrotchett )

Update: After Dave's feedback about the issue with duplicated named ideas, I've tried again today and I successfully submitted our idea for staging! I agree with him also that there's a pop-up during submission that might be confusing but we've met the acceptance criteria for this ticket, so we are able to close it.

Thank you so much @RETprojects  and @dbradham  for your hard work! 

<img width="1332" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBMmViQlE9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--c420f0055287d6352547b38b67e263928694bb7c/image.png" alt="image.png" />

<img width="1341" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBMnFiQlE9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--cbe505567d530df4ca2488a70bb0f8c0c8eea1aa/image.png" alt="image.png" />
